packages/xo-web/src/xo-app/home/index.js

```diff
@@ -439,26 +446,43 @@ const NoObjects = props =>
   props.isAdmin ? <NoObjectsWithServers {...props} /> : <NoObjectsWithoutServers {...props} />
 
 @connectStore(() => {
-  const type = (_, props) => props.location.query.t || DEFAULT_TYPE
+  const getType = (_, props) => props.location.query.t || DEFAULT_TYPE
+  const getContainers = createSelector(
+    createGetObjectsOfType('host'),
+    createGetObjectsOfType('pool'),
+    (hosts, pools) => ({ ...hosts, ...pools })
+  )
+  const getItems = createSelector(getContainers, createGetObjectsOfType(getType), (containers, items) =>
+    mapValues(items, item => ({
+      ...item,
+      container: containers[item.$container || item.$pool],
+    }))
+  )
 
-  return {
-    areObjectsFetched,
-    isAdmin,
-    isPoolAdmin: getIsPoolAdmin,
-    items: createSelector(
-      createSelector(createGetObjectsOfType('host'), createGetObjectsOfType('pool'), (hosts, pools) => ({
-        ...hosts,
-        ...pools,
-      })),
-      createGetObjectsOfType(type),
-      (containers, items) =>
-        mapValues(items, item => ({
-          ...item,
-          container: containers[item.$container || item.$pool],
-        }))
-    ),
-    type,
-    user: getUser,
+  const getVms = createSelector(
```

Comment:
Add a comment so that we quickly understand why you need a separate selector for VMs.